pov/pov_test.go

```diff
@@ -0,0 +1,317 @@
+			{"aunt", "cousin-2"},
+			{"grand-parent", "parent"},
+			{"grand-parent", "aunt"},
+		},
+		root: "grand-parent",
+		arcStrings: []string{
+			"aunt -> cousin-1",
+			"aunt -> cousin-2",
+			"grand-parent -> aunt",
+			"grand-parent -> parent",
+			"parent -> sib-1",
+			"parent -> sib-2",
+			"parent -> x",
+		},
+		reRooted: []string{
+			"aunt -> cousin-1",
+			"aunt -> cousin-2",
+			"grand-parent -> aunt",
+			"parent -> grand-parent",
+			"parent -> sib-1",
+			"parent -> sib-2",
+			"x -> parent",
+		},
+	},
+	{
+		description: "target with children",
+		leaves: []string{"child-1", "child-2", "nephew", "niece",
+			"2nd-cousin-1", "2nd-cousin-2", "2nd-cousin-3", "2nd-cousin-4"},
+		arcPairs: []arc{
+			{"x", "child-1"},
+			{"x", "child-2"},
+			{"sibling", "nephew"},
+			{"sibling", "niece"},
+			{"cousin-1", "2nd-cousin-1"},
+			{"cousin-1", "2nd-cousin-2"},
+			{"cousin-2", "2nd-cousin-3"},
+			{"cousin-2", "2nd-cousin-4"},
+			{"parent", "x"},
+			{"parent", "sibling"},
+			{"aunt", "cousin-1"},
+			{"aunt", "cousin-2"},
+			{"grand-parent", "parent"},
+			{"grand-parent", "aunt"},
+		},
+		root: "grand-parent",
+		arcStrings: []string{
+			"aunt -> cousin-1",
+			"aunt -> cousin-2",
+			"cousin-1 -> 2nd-cousin-1",
+			"cousin-1 -> 2nd-cousin-2",
+			"cousin-2 -> 2nd-cousin-3",
+			"cousin-2 -> 2nd-cousin-4",
+			"grand-parent -> aunt",
+			"grand-parent -> parent",
+			"parent -> sibling",
+			"parent -> x",
+			"sibling -> nephew",
+			"sibling -> niece",
+			"x -> child-1",
+			"x -> child-2",
+		},
+		reRooted: []string{
+			"aunt -> cousin-1",
+			"aunt -> cousin-2",
+			"cousin-1 -> 2nd-cousin-1",
+			"cousin-1 -> 2nd-cousin-2",
+			"cousin-2 -> 2nd-cousin-3",
+			"cousin-2 -> 2nd-cousin-4",
+			"grand-parent -> aunt",
+			"parent -> grand-parent",
+			"parent -> sibling",
+			"sibling -> nephew",
+			"sibling -> niece",
+			"x -> child-1",
+			"x -> child-2",
+			"x -> parent",
+		},
+	},
+}
+
+func same(a, b []string) bool {
+	return len(a)+len(b) == 0 || reflect.DeepEqual(a, b)
+}
+
+func TestConstruction(t *testing.T) {
+	if TestVersion != testVersion {
+		t.Fatalf("Found TestVersion = %v, want %v", TestVersion, testVersion)
+	}
+	// very similar code below in TestChangeRoot
+	for _, tc := range testCases {
+		g := New()
+		for _, l := range tc.leaves {
+			g.AddNode(l)
+		}
+		for _, a := range tc.arcPairs {
+			g.AddArc(a.fr, a.to)
+		}
+		got := g.ArcList()
+		gs := append([]string{}, got...)
+		sort.Strings(gs)
+		if !same(gs, tc.arcStrings) {
+			t.Logf("got %d arcs:", len(got))
+			for _, s := range got {
+				t.Log(" ", s)
+			}
+			t.Logf("that result sorted:")
+			for _, s := range gs {
+				t.Log(" ", s)
+			}
+			t.Logf("want %d arcs:", len(tc.arcStrings))
+			for _, s := range tc.arcStrings {
+				t.Log(" ", s)
+			}
+			t.Fatal("incorrect graph construction")
+		}
+	}
+}
+
+func TestChangeRoot(t *testing.T) {
+	// very similar code above in TestConstruction
+	for _, tc := range testCases {
+		g := New()
```

Comment:
given that L261-L267 are the same as in above, I might suggest adding a function on the `testCase` struct that generates the graph from it, to avoid the code repetition.

(obviously that would need `type testCase struct { ... }` rather than using the anonymous struct as it currently is)